cmd/rig/cmd/capsule/root/port-forward.go

```diff
@@ -0,0 +1,277 @@
+			var choices []string
+			for _, i := range spec.GetInterfaces() {
+				choices = append(choices, fmt.Sprintf("%s (port %d)", i.GetName(), i.GetPort()))
+			}
+
+			i, _, err := c.Prompter.Select("Select a network interface", choices)
+			if err != nil {
+				return err
+			}
+
+			remotePort = uint32(spec.GetInterfaces()[i].GetPort())
+		} else {
+			remotePort = uint32(spec.GetInterfaces()[0].GetPort())
+		}
+
+		localPort = remotePort
+	}
+
+	if instanceID == "" {
+		instancesRes, err := c.Rig.Capsule().ListInstances(ctx, &connect.Request[capsule.ListInstancesRequest]{
+			Msg: &capsule.ListInstancesRequest{
+				ProjectId:     c.Scope.GetCfg().GetProject(),
+				EnvironmentId: c.Scope.GetCfg().GetEnvironment(),
+				CapsuleId:     capsuleID,
+				Pagination: &model.Pagination{
+					Limit: 1,
+				},
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		if len(instancesRes.Msg.Instances) == 0 {
+			return errors.NotFoundErrorf("no instances found for capsule")
+		}
+
+		instanceID = instancesRes.Msg.Instances[0].GetInstanceId()
+	}
+
+	l, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", localPort))
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("[rig] connected to instance '%s', accepting traffic on %s\n", instanceID, l.Addr().String())
+
+	if follow {
+		go func() {
+			for {
+				time.Sleep(1 * time.Second)
+
+				res, err := c.Rig.Capsule().Logs(ctx, &connect.Request[capsule.LogsRequest]{
+					Msg: &capsule.LogsRequest{
+						ProjectId:     c.Scope.GetCfg().GetProject(),
+						EnvironmentId: c.Scope.GetCfg().GetEnvironment(),
+						CapsuleId:     capsuleID,
+						InstanceId:    instanceID,
+						Follow:        true,
+						Since:         durationpb.New(10 * time.Second),
+					},
+				})
+				if err != nil {
+					fmt.Printf("[rig] error tailing logs: %v\n", err)
+					continue
+				}
+
+				for res.Receive() {
+					switch v := res.Msg().GetLog().GetMessage().GetMessage().(type) {
+					case *capsule.LogMessage_ContainerTermination_:
+						fmt.Printf("[rig] instance restarted")
+						res.Close()
+					case *capsule.LogMessage_Stdout:
+						os.Stdout.Write(v.Stdout)
+					case *capsule.LogMessage_Stderr:
+						os.Stdout.Write(v.Stderr)
+					}
+				}
+
+				if res.Err() != nil {
+					fmt.Printf("[rig] error tailing logs: %v\n", err)
+				}
+
+				res.Close()
+			}
+		}()
+	}
+
+	for {
+		conn, err := l.Accept()
+		if err != nil {
+			if verbose {
+				return fmt.Errorf("error listening for incoming connections: %v", err)
+			}
+		}
+
+		if verbose {
+			fmt.Printf("[rig] new connection %s -> %s:%d\n", conn.RemoteAddr().String(), instanceID, remotePort)
+		}
+
+		go func() {
+			err := runPortForward(ctx, capsuleID, instanceID, conn, c.Scope, c.Rig, remotePort)
+			if err != nil {
+				fmt.Println("[rig] connection closed with error:", err)
+			} else if verbose {
+				fmt.Printf("[rig] closed connection %s -> %s:%d\n", conn.RemoteAddr().String(), instanceID, remotePort)
+			}
+		}()
+	}
+}
+
+func runPortForward(
```

Comment:
Make instance method on `Cmd`, then you don't need to send `Scope` and `Rig` as arguments